File: backend/app/schemas/order.py
from pydantic import BaseModel
import uuid
from datetime import datetime

class OrderItemCreate(BaseModel):
    product_id: uuid.UUID
    quantity: int

class OrderCreateRequest(BaseModel):
    phone: str
    business_id: uuid.UUID
    items: list[OrderItemCreate]
    delivery_lat: float
    delivery_lon: float

class OrderItemResponse(BaseModel):
    id: uuid.UUID
    product_name: str
    unit_price: float
    quantity: int
    product_id: uuid.UUID | None
    thumbnail_url: str | None = None

class OrderResponse(BaseModel):
    distance_km: float | None = None
    delivery_location: dict | None = None
    delivery_coordinates: dict | None = None
    customer_phone: str | None = None
    id: uuid.UUID
    order_number: str
    status: str
    subtotal: float
    delivery_fee: float
    total_amount: float
    customer_id: uuid.UUID
    business_id: uuid.UUID
    business_name: str | None = None
    business_logo_url: str | None = None
    business_cover_url: str | None = None
    items: list[OrderItemResponse] = []
    created_at: datetime | None

    class Config:
        from_attributes = True

File: frontend/apps/rider/src/types/order.ts
export interface OrderItem {
  id: string;
  product_name: string;
  unit_price: number;
  quantity: number;
  thumbnail_url: string | null;
}

export interface Order {
  id: string;
  order_number: string;
  status: string;
  subtotal: number;
  delivery_fee: number;
  total_amount: number;
  customer_id: string;
  business_id: string;
  customer_name?: string | null;
  customer_phone?: string | null;
  business_name?: string | null;
  business_logo?: string | null;
  pickup_location?: { lat: any; lon: any } | null;
  delivery_location?: { lat: any; lon: any } | null;
  items: OrderItem[];
}
